core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -87,6 +108,34 @@ public static BaseFileSystem create(FileSystemContext context) {
    */
   protected BaseFileSystem(FileSystemContext fsContext) {
     mFsContext = fsContext;
+    mBlockStore = AlluxioBlockStore.create(fsContext);
+  }
+
+  /**
+   * Shuts down the FileSystem. Closes all thread pools and resources used to perform operations. If
+   * any operations are called after closing the context the behavior is undefined.
+   *
+   * @throws IOException
+   */
+  @Override
+  public void close() throws IOException {
```

Comment:
are we allowed to call close() multiple times?

Reply:
Ideally the user shouldn't but if they do, the javadoc interfaces specifies

Reply:
So I'm choosing to implement invoking `close()` a second time as a no-op